tests/imputers/test_NearestMeanResponseImputer.py

```diff
@@ -160,161 +158,12 @@ def test_learnt_values2(self):
-        df = create_NearestMeanResponseImputer_test_df()
-
-        x = NearestMeanResponseImputer(columns=["a", "b"])
-
-        x.fit(df, df["c"])
-
-        expected_call_args = {
-            0: {"args": (create_NearestMeanResponseImputer_test_df(),), "kwargs": {}},
-        }
-
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "transform",
-            expected_call_args,
-        ):
-            x.transform(df)
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.adjusted_dataframe_params(
-            create_NearestMeanResponseImputer_test_df(),
-            expected_df_1(),
-        ),
-    )
-    def test_nulls_imputed_correctly(self, df, expected):
-        """Test missing values are filled with the correct values."""
-        x = NearestMeanResponseImputer(columns=["a", "b"])
-
-        # set the impute values dict directly rather than fitting x on df so test works with helpers
-        x.impute_values_ = {"a": 2.0, "b": 3.0}
-
-        df_transformed = x.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg="Check nulls filled correctly in transform",
-        )
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.adjusted_dataframe_params(
-            create_NearestMeanResponseImputer_test_df(),
-            expected_df_2(),
-        ),
-    )
-    def test_nulls_imputed_correctly2(self, df, expected):
-        """Test missing values are filled with the correct values - and unrelated columns are unchanged."""
-        x = NearestMeanResponseImputer(columns="a")
-
-        # set the impute values dict directly rather than fitting x on df so test works with helpers
-        x.impute_values_ = {"a": 2.0}
-
-        df_transformed = x.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg="Check nulls filled correctly in transform",
-        )
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.adjusted_dataframe_params(
-            pd.DataFrame({"a": [np.nan, 3, 4, 1, 4, np.nan]}),
-            expected_df_3(),
-        ),
-    )
-    def test_nulls_imputed_correctly3(self, df, expected):
-        """Test missing values are filled with the correct values - with median value from separate dataframe."""
-        x = NearestMeanResponseImputer(columns="a")
-
-        # set the impute values dict directly rather than fitting x on df so test works with helpers
-        x.impute_values_ = {"a": 2.0}
-
-        df_transformed = x.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg="Check nulls filled correctly in transform",
-        )
-
-    def test_learnt_values_not_modified(self):
```

Comment:
Q: do we not need this one?

Reply:
Hey Liam - is this not covered by the 'test_impute_value_unchanged' test in the 'test_BaseImputer' script?